Description: fix VNC platform plugin build on big-endian machines
Author: Shawn Rutledge <shawn.rutledge@digia.com>
Origin: https://codereview.qt-project.org/185248
Last-Update: 2017-02-16

--- a/src/plugins/platforms/vnc/qvncclient.cpp
+++ b/src/plugins/platforms/vnc/qvncclient.cpp
@@ -142,7 +142,7 @@
         case 16: {
             quint16 p = *reinterpret_cast<const quint16*>(src);
 #if Q_BYTE_ORDER == Q_BIG_ENDIAN
-            if (swapBytes)
+            if (m_swapBytes)
                 p = ((p & 0xff) << 8) | ((p & 0xff00) >> 8);
 #endif
             r = (p >> 11) & 0x1f;
@@ -484,7 +484,7 @@
         m_sameEndian = (QSysInfo::ByteOrder == QSysInfo::BigEndian) == !!m_pixelFormat.bigEndian;
         m_needConversion = pixelConversionNeeded();
 #if Q_BYTE_ORDER == Q_BIG_ENDIAN
-        m_swapBytes = qvnc_screen->swapBytes();
+        m_swapBytes = server()->screen()->swapBytes();
 #endif
     }
 }
@@ -639,7 +639,7 @@
         return true;
 
 #if Q_BYTE_ORDER == Q_BIG_ENDIAN
-    if (qvnc_screen->swapBytes())
+    if (server()->screen()->swapBytes())
         return true;
 #endif
 
--- a/src/plugins/platforms/vnc/qvncscreen.cpp
+++ b/src/plugins/platforms/vnc/qvncscreen.cpp
@@ -43,6 +43,7 @@
 #include <QtFbSupport/private/qfbcursor_p.h>
 
 #include <QtGui/QPainter>
+#include <QtGui/QScreen>
 #include <QtCore/QRegularExpression>
 
 
@@ -172,14 +173,18 @@
 }
 
 #if Q_BYTE_ORDER == Q_BIG_ENDIAN
-bool QVNCScreen::swapBytes() const
+bool QVncScreen::swapBytes() const
 {
+    return false;
+
+    /* TODO
     if (depth() != 16)
         return false;
 
     if (screen())
         return screen()->frameBufferLittleEndian();
     return frameBufferLittleEndian();
+    */
 }
 #endif
 
--- a/src/plugins/platforms/vnc/qvncscreen.h
+++ b/src/plugins/platforms/vnc/qvncscreen.h
@@ -73,7 +73,7 @@
     void clearDirty() { dirtyRegion = QRegion(); }
 
 #if Q_BYTE_ORDER == Q_BIG_ENDIAN
-    bool swapBytes() const
+    bool swapBytes() const;
 #endif
 
     QStringList mArgs;
